### PR TITLE
refactor: use styled components in Badge

### PR DESCRIPTION
**Pull request by LeandroTorresSicilia**
## Changes proposed in this PR:
- use styled components in Badge

- [ ] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

File: src/components/Badge/truncate.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import styled from 'styled-components';
+
+const Truncate = styled.span`
```
Comment (reiniergs) [inline src/components/Badge/truncate.js]: TruncatedText??

**Comment by gustavopch** [comment]
In case you don't know, you don't need to import those theme variables. Instead, you could do something like:
```
color: ${props => props.theme.COLOR_GRAY_4}
```
Of course you'd have to change the Application component to have a `ThemeProvider`. Then, when lib is fully refactored with styled-components, it'll be a matter of changing Application to take a `theme` prop in order to let each consumer customize their themes or implement a dark mode.

**Comment by LeandroTorresSicilia** [comment]
@gustavopch 
That's right, but I think first we should move forward refactoring all components to use styled-components and removing all scss files.
For theming we need to discuss its structure and what will be customizable so it is early to use:
```props.theme.COLOR_GRAY_4 ```
although I know that we will need to touch all components again

**Comment by TahimiLeonBravo** [comment]
#950

**Pull request merged**